resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -189,6 +178,17 @@ private[spark] class KubernetesClusterSchedulerBackend(
           .withContainerPort(port._2)
           .build()
       })
+    val args = scala.collection.mutable.Buffer[String]()
+    args ++= Seq(
+      "${JAVA_HOME}/bin/java", s"-Dspark.executor.port=${executorPort.toString}",
```

Comment:
It's exactly what I wanted to avoid is to let the Dockerfile specify the main class, then it becomes coupled with the Dockerfile with a particular version.